### PR TITLE
MySQL CDC supports parsing pt-osc generated schema changes

### PR DESCRIPTION
**Pull request by MOBIN-F**
MySQL CDC supports parsing pt-osc generated schema changes

### CODE REVIEW

**Comment by MOBIN-F** [comment]
@yuxiqian I have completed the rest of the work on pt-osc. The main changes are as follows. Can you help review it?
1. There is no identifier like /* gh-ost*/ in the change record of pt-osc, so I modified some logic
2. Both pt-osc and gh-ost belong to the Online Schema Change tool, so I renamed some of them: XXXGhOstSchemaChanges -> XXXOnLineSchemaChanges

File: flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mysql-cdc/src/main/java/org/apache/flink/cdc/connectors/mysql/source/config/MySqlSourceOptions.java [hunk 1]
```diff
@@ -264,8 +264,8 @@ public class MySqlSourceOptions {
                             "Whether to skip backfill in snapshot reading phase. If backfill is skipped, changes on captured tables during snapshot phase will be consumed later in binlog reading phase instead of being merged into the snapshot. WARNING: Skipping backfill might lead to data inconsistency because some binlog events happened within the snapshot phase might be replayed (only at-least-once semantic is promised). For example updating an already updated value in snapshot, or deleting an already deleted entry in snapshot. These replayed binlog events should be handled specially.");
 
     @Experimental
-    public static final ConfigOption<Boolean> PARSE_GH_OST_SCHEMA_CHANGES =
-            ConfigOptions.key("parse.gh-ost.schema.changes")
+    public static final ConfigOption<Boolean> PARSE_ONLINE_SCHEMA_CHANGES =
+            ConfigOptions.key("parse.online.schema.changes")
                     .booleanType()
                     .defaultValue(false)
                     .withDescription(
```
Comment (yuxiqian) [inline flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mysql-cdc/src/main/java/org/apache/flink/cdc/connectors/mysql/source/config/MySqlSourceOptions.java:271]: `description` needs to be updated, too

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/java/org/apache/flink/cdc/connectors/mysql/source/MySqlOnLineSchemaMigrationITCase.java [hunk 2]
```diff
@@ -310,6 +331,192 @@ public void testGhOstSchemaMigrationFromScratch() throws Exception {
                 fetchResults(events, 2));
     }
 
+    @Test
+    public void testPtOscSchemaMigrationFromScratch() throws Exception {
+        LOG.info("Step 2: Start pipeline job");
+        //        Thread.sleep(5000000);
```
Comment (yuxiqian) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/java/org/apache/flink/cdc/connectors/mysql/source/MySqlOnLineSchemaMigrationITCase.java]:
```suggestion

```

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/java/org/apache/flink/cdc/connectors/mysql/source/MySqlOnLineSchemaMigrationITCase.java [hunk 3]
```diff
@@ -310,6 +331,192 @@ public void testGhOstSchemaMigrationFromScratch() throws Exception {
...
+        env.setParallelism(1);
+        customerDatabase.createAndInitialize();
+        TableId tableId = TableId.tableId(customerDatabase.getDatabaseName(), "customers_1");
```
Comment (yuxiqian) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/java/org/apache/flink/cdc/connectors/mysql/source/MySqlOnLineSchemaMigrationITCase.java]: We may avoid table name conflicts by creating & destroying database in `@Before` and `@After` method for each test case

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/java/org/apache/flink/cdc/connectors/mysql/source/MySqlOnLineSchemaMigrationITCase.java [hunk 4]
```diff
@@ -310,6 +331,192 @@ public void testGhOstSchemaMigrationFromScratch() throws Exception {
                 fetchResults(events, 2));
     }
 
+    @Test
+    public void testPtOscSchemaMigrationFromScratch() throws Exception {
+        LOG.info("Step 2: Start pipeline job");
```
Comment (yuxiqian) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/java/org/apache/flink/cdc/connectors/mysql/source/MySqlOnLineSchemaMigrationITCase.java]: Seems there's no step 1

**Review by yuxiqian — COMMENTED** [review]
Thanks for Mobin's quick fix! Just left some minor comments.

File: flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mysql-cdc/src/main/java/org/apache/flink/cdc/connectors/mysql/source/config/MySqlSourceOptions.java (hunk 1 above)
Comment (MOBIN-F) [reply flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mysql-cdc/src/main/java/org/apache/flink/cdc/connectors/mysql/source/config/MySqlSourceOptions.java:271]: done

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/java/org/apache/flink/cdc/connectors/mysql/source/MySqlOnLineSchemaMigrationITCase.java (hunk 2 above)
Comment (MOBIN-F) [reply flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/java/org/apache/flink/cdc/connectors/mysql/source/MySqlOnLineSchemaMigrationITCase.java]: done

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/java/org/apache/flink/cdc/connectors/mysql/source/MySqlOnLineSchemaMigrationITCase.java (hunk 4 above)
Comment (MOBIN-F) [reply flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/java/org/apache/flink/cdc/connectors/mysql/source/MySqlOnLineSchemaMigrationITCase.java]: done

**Comment by yuxiqian** [comment]
Thanks for helping on this!

**Pull request merged**